File: 1_two_sum.py
def twoSum(self, nums, target):
    seen = {}
    for idx, num in enumerate(nums, 0):
        if seen.get(target - num) == 0 or seen.get(target - num):
            return [seen[target - num], idx]
        elif not seen.get(nums[idx]):
            seen[num] = idx

File: leetcode/1_two_sum.js
var twoSum = function(nums, target) {
  let seen = {};
  for (let i = 0; i < nums.length; i++) {
    // have to check index 0 specifically cause it's falsey
    if (seen[target - nums[i]] === 0 || seen[target - nums[i]]) {
      return [seen[target - nums[i]], i];
    } else if (!seen[nums[i]]) {
      seen[nums[i]] = i;
    }
  }
};
